onesky.go

```diff
@@ -58,11 +61,178 @@ type LastImport struct {
+	CreateddAtTimestamp int         `json:"created_at_timestamp"`
+}
+
+// Language is a struct which contains informations about locale
+type Language struct {
+	Code         string `json:"code"`
+	EnglishName  string `json:"english_name"`
+	LocalName    string `json:"local_name"`
+	CustomLocale string `json:"custom_locale"`
+	Locale       string `json:"locale"`
+	Region       string `json:"region"`
+}
+
+// TaskFile is a struct which contains informations about file of import task
+type TaskFile struct {
+	Name   string   `json:"name"`
+	Format string   `json:"format"`
+	Locale Language `json:"locale"`
+}
+
+// ImportTasksResponse is a struct which contains informations about the response from list import tasks API
+type ImportTasksResponse struct {
+	Data []TaskData `json:"data"`
+}
+
+// ImportTaskResponse is a struct which contains informations about the response from show an import task API
+type ImportTaskResponse struct {
+	Data TaskData `json:"data"`
+}
+
+// UploadData is a struct which contains informations about uploaded file
+type UploadData struct {
+	Name     string   `json:"name"`
+	Format   string   `json:"format"`
+	Language Language `json:"language"`
+	Import   TaskData `json:"import"`
+}
+
+// UploadResponse is a struct which contains informations about the response from upload file API
+type UploadResponse struct {
+	Data UploadData `json:"data"`
+}
+
+// ConvertInt : Convert interface{} to int
+func ConvertInt(in interface{}) (int64, error) {
+	switch in.(type) {
+	case string:
+		return strconv.ParseInt(in.(string), 10, 64)
+	case int:
+		return int64(in.(int)), nil
+	case int16:
+		return int64(in.(int16)), nil
+	case int32:
+		return int64(in.(int32)), nil
+	case int64:
+		return in.(int64), nil
+	case uint:
+		return int64(in.(uint)), nil
+	case uint16:
+		return int64(in.(uint16)), nil
+	case uint32:
+		return int64(in.(uint32)), nil
+	case uint64:
+		return int64(in.(uint64)), nil
+	case float32:
+		return int64(in.(float32)), nil
+	case float64:
+		return int64(in.(float64)), nil
+	default:
+		return 0, fmt.Errorf("%s: %v", "Unable to convert value", in)
+	}
+}
+
+// ImportTask : Show an import task. Parameters: import_id
+func (c *Client) ImportTask(importID int64) (TaskData, error) {
+	endpoint, err := getEndpoint("importTask")
+	if err != nil {
+		return TaskData{}, err
+	}
+	values := url.Values{}
+	urlStr, err := endpoint.full(c, values, importID)
+	if err != nil {
+		return TaskData{}, err
+	}
+
+	res, err := makeRequest(endpoint.method, urlStr, nil, "")
+	if err != nil {
+		return TaskData{}, err
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return TaskData{}, fmt.Errorf("bad status: %s", res.Status)
+	}
+
+	body, err := getResponseBodyAsString(res)
+	if err != nil {
+		return TaskData{}, err
+	}
+	aux := ImportTaskResponse{}
+	err = json.Unmarshal([]byte(body), &aux)
+	if err != nil {
+		return TaskData{}, err
+	}
+	if i, err := ConvertInt(aux.Data.OriginalID); err == nil {
+		// fmt.Printf("%T, %v", i, i)
```

Comment:
Do you mean line 180?
`if i, err := ConvertInt(aux.Data.OriginalID); err == nil {`
Because onesky's API is non-standard, ID may be string or int.